components/markets/SimilarMarketsSection.tsx

```diff
@@ -0,0 +1,94 @@
+import type { FullMarketFragment } from "@zeitgeistpm/indexer";
+import { ScalarRangeType } from "@zeitgeistpm/sdk-next";
+import MarketCard from "components/markets/market-card";
+import Decimal from "decimal.js";
+import { ZTG } from "lib/constants";
+import { useMarketsStats } from "lib/hooks/queries/useMarketsStats";
+import { useSimilarMarkets } from "lib/hooks/queries/useSimilarMarkets";
+
+export const SimilarMarketsSection = ({
+  market,
+  limit,
+}: {
+  market?: FullMarketFragment;
+  limit?: number;
+}) => {
+  const similarMarkets = useSimilarMarkets(market?.marketId, limit ?? 2);
+
+  const hasSimilarMarkets = Boolean(similarMarkets?.data?.length);
+
+  const stats = useMarketsStats(
+    similarMarkets?.data?.map((m) => m.marketId) ?? [],
+    {
+      enabled: hasSimilarMarkets,
+    },
+  );
+
+  const isLoading =
```

Comment:
I think this can be simplified quite a bit. Think if `isFetched` is true you can assume it's not loading anymore. https://tanstack.com/query/v4/docs/react/reference/useQuery